Frameworks/Foundation/NSUserDefaults.mm

```diff
@@ -48,6 +48,7 @@
 
 @implementation NSUserDefaults {
     StrongId<NSMutableDictionary> _registrationDict;
+    StrongId<NSOperationQueue> _synchronizeQueue;
 }
```

Comment:
A quick examination of our NSOperationQueue implementation shows that we destroy the worker thread whenever the queue is empty and create a new thread when an operation is added. That's inefficient compared to using threadpools, but we should make the fix in NSOperationQueue rather than switching to a threadpool in NSUserDefaults because:

1) The performance gains in NSUserDefaults would be nominal compared to the already huge cost of writing to file
2) We would now need to write two fixes instead of one and maintain two threadpool/queue implementations instead of one.

---
In reply to: [69818599](https://github.com/Microsoft/WinObjC/pull/612#discussion_r69818599) [](ancestors = 69818599,69786383)